src/main/java/com/stuypulse/robot/subsystems/Conveyor.java

```diff
@@ -42,96 +39,130 @@
     }
-    
+
     /**
-     * Eject ball - spin the gandalf motor outwards
-     * To be used when the ball is opposing alliance color
+     * Eject ball - spin the gandalf motor outwards To be used when the ball is opposing alliance
+     * color
      */
-    public void rejectBall(){
+    private void rejectBall() {
         // If the ball is not of our alliance color, reject ball
         gandalfMotor.set(ConveyorSettings.REJECT_SPEED.get());
     }
-    
-    /**
-     * Stops the Top Conveyor Belt
-     */
-    public void stopTopBelt() {
+
+    /** Stops the Top Conveyor Belt */
+    private void stopTopBelt() {
         topMotor.stopMotor();
     }
 
-    /**
-     * Stops the Gandalf Motor
-     */
-    public void stopGandalf() {
+    /** Stops the Gandalf Motor */
+    private void stopGandalf() {
         gandalfMotor.stopMotor();
     }
 
-    /**
-     * Stop both the Top Conveyor Belt and the Gandalf Motor
-     */
-    public void stop() {
-        stopTopBelt();
-        stopGandalf();
+    /** Finds if the upper IR Sensor has been tripped e.g., there is a ball in the top conveyor */
+    private boolean getTopConveyorUpperHasBall() {
+        return topIRSensor.get();
     }
 
-    /**
-     * Gets the CurrentBall object from the color sensor
-     * @return CurrentBall detected by the color sensor
-     */
-    private CurrentBall getCurrentBall() {
-        return colorSensor.getCurrentBall();
+    /** Finds if the lower IR Sensor has been tripped */
+    private boolean getTopConveyorLowerHasBall() {
+        return gandalfIRSensor.get();
     }
-    
-    /**
-     * Finds if the upper IR Sensor has been tripped e.g., there is a ball in the top conveyor
-     * @return if the upper IR Sensor has been tripped
-     */
-    public boolean getTopConveyorHasBall() {
-        return irSensor.get();
+
+    private boolean getEjecting() {
+        return colorSensor.hasOpponentBall();
+    }
+
+    private boolean getRunning() {
+        if (isShooting()) { // same as 3rd?
+            return true;
+        } else if (getTopConveyorUpperHasBall()) { // top IR
+            return false; // all motors are stopped according to logic tabl
+        } else if (colorSensor.hasAllianceBall()) { // good gap
+            return true;
+        } else {
+            return false;
+        }
     }
 
     @Override
     public void periodic() {
+        // boolean bottomIRHasBall = getTopConveyorLowerHasBall();
+        // boolean topIRHasBall = getTopConveyorUpperHasBall();
+
+        boolean ejecting = getEjecting();
+        boolean running = getRunning();
+
+        // Gandalf
+        if (ejecting) {
+            rejectBall();
+        } else if (running) {
+            acceptBall();
+        } else {
+            stopGandalf();
+        }
+
+        // Top Conveyor Motor
+        if (running) {
+            spinTopBelt();
+        } else {
+            stopTopBelt();
+        }
```

Comment:
we need to consider if we want this code this code to be a command or in periodic.

Pros to Periodic:
- cleaner subsystem
- less public functions
- simpler behavior

Pros to Commands:
- More modular behavior (can switch one logic command out for another)
- Forces you to clean up the Conveyor API a bit more
- Downside is increased complexity

I am a fan of doing this in Periodic, but I worry what will happen as we add in edge cases like disabling ejection or force ejecting balls.